### PR TITLE
Adds minimum version requirement for kubectl

### LINKED ISSUE #8341: BDC needs kubectl version >=1.13
Based on this documentation: https://docs.microsoft.com/en-us/sql/big-data-cluster/deploy-big-data-tools?view=sql-server-ver15#big-data-cluster-tools

kubectl needed for BDC deployment needs to be at least version 1.13 so we need to add this check to our BDC deployment tools acquisition story.

### PR DESCRIPTION
**Pull request by ranasaria**
This PR fixes #8341
1. Add version requirement for Kubectl in package.config
2. Fix a bug introduced previously where if wrong version is installed the 'installTools' button was no longer disabled.

### CODE REVIEW

File: extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts [hunk 1]
```diff
@@ -235,6 +235,11 @@ export class ResourceTypePickerDialog extends DialogBase {
 					this._toDispose.push(tool.onDidUpdateData((t: ITool) => {
 						this.updateToolsDisplayTableData(t);
 					}));
+					console.log(`TCL: tool`, tool);
```
Comment (alanrenmsft) [inline extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts]: I'd not add these

**Comment by Charles-Gagnon** [comment]
General comment - but please try to make the titles more descriptive so it's easy to see what is happening just from looking at the title.

File: extensions/resource-deployment/package.json [hunk 2]
```diff
@@ -248,7 +248,8 @@
 						},
 						"requiredTools": [
 							{
-								"name": "kubectl"
+                "name": "kubectl",
+                "version": "1.13.0"
 							},
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/package.json]: spaces -> tabs

File: extensions/resource-deployment/src/services/tools/toolBase.ts [hunk 3]
```diff
@@ -100,8 +100,8 @@ export abstract class ToolBase implements ITool {
 		return <string>toolStatusLocalized.get(this._status);
 	}
 
-	public get autoInstallRequired(): boolean {
-		return this.status !== ToolStatus.Installed && this.autoInstallSupported;
+	public isInstallRequired(minimumVersion?: string): boolean {
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/services/tools/toolBase.ts]: >isInstallRequired [](start = 8, length = 17)

I'd suggest renaming this to canAutoInstall or something similar - this isn't really saying anything about the install being "required" in my eyes. Just that we can install a valid version.

File: extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts (hunk 1 above)
Comment (ranasaria) [reply extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts]: removed already

File: extensions/resource-deployment/src/services/tools/toolBase.ts (hunk 3 above)
Comment (ranasaria) [reply extensions/resource-deployment/src/services/tools/toolBase.ts]: will rename to isMissingOrOld

**Pull request merged**